docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-conditions-autodiscover.asciidoc

```diff
@@ -0,0 +1,290 @@
+------------------------------------------------
+- name: redis
+  type: redis/metrics
+  use_output: default
+  meta:
+    package:
+      name: redis
+      version: 0.3.6
+  data_stream:
+    namespace: default
+  streams:
+    - data_stream:
+        dataset: redis.info
+        type: metrics
+      metricsets:
+        - info
+      hosts:
+        - '${kubernetes.pod.ip}:6379'
+      idle_timeout: 20s
+      maxconn: 10
+      network: tcp
+      period: 10s
+      condition: ${kubernetes.labels.app} == 'redis'
+------------------------------------------------
+
+The condition `${kubernetes.labels.app} == 'redis'` will make the {agent} look for a Pod with the  label `app:redis` within the scope defined in its manifest.
+
+For a list of provider fields that you can use in conditions, refer to <<kubernetes-provider>>.
+
+
+The `redis` input defined in the {agent} manifest only specifies the`info` metricset. To learn about other available metricsets and their configuration settings, refer to the {metricbeat-ref}/metricbeat-module-redis.html[Redis module page].
+
+To deploy Redis, you can apply the following example manifest:
+
+[source,yaml]
+------------------------------------------------
+apiVersion: v1
+kind: Pod
+metadata:
+  name: redis
+  annotations:
+    co.elastic.hints/package: redis
+    co.elastic.hints/host: '${kubernetes.pod.ip}:6379'
+  labels:
+    k8s-app: redis
+    app: redis
+spec:
+  containers:
+  - image: redis
+    imagePullPolicy: IfNotPresent
+    name: redis
+    ports:
+    - name: redis
+      containerPort: 6379
+      protocol: TCP
+------------------------------------------------
+
+You should now be able to see Redis data flowing in on index `metrics-redis.info-default`. Make sure the port in your Redis manifest file matches the port used in the Redis input.
+
+NOTE: All assets (dashboards, ingest pipelines, and so on) related to the Redis integration are not installed. You need to explicitly <<install-uninstall-integration-assets,install them through {kib}>>.
+
+To set the target host dynamically for a targeted Pod based on its labels, use a variable in the {agent} policy to return path information from the provider:
+
+[source,yaml]
+----
+- data_stream:
+    dataset: kubernetes.scheduler
+    type: metrics
+  metricsets:
+    - scheduler
+  bearer_token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
+  hosts:
+    - 'https://${kubernetes.pod.ip}:10259'
+  period: 10s
+  ssl.verification_mode: none
+  condition: ${kubernetes.labels.component} == 'kube-scheduler'
+----
+
+WARNING: In some "As a Service" Kubernetes implementations, like GKE, the control plane nodes or even the Pods running on them won’t be visible. In these cases, it won’t be possible to use scheduler metricsets, necessary for this example. Refer https://www.elastic.co/guide/en/beats/metricbeat/current/metricbeat-module-kubernetes.html#_scheduler_and_controllermanager[scheduler and controller manager] to find more information.
```

Comment:
I think it is important because readers get an idea how to set the host dynamically, and if we don't include it they may not get there on their own. I can only speak for myself, but as someone with little experience with Kubernetes, it would took me awhile to get there.

Reply:
At first it was indeed a separate block. But after discussion, [it changed ](https://github.com/elastic/observability-docs/pull/2366#discussion_r1024087341)

Reply:
I still don't understand why example with scheduler is needed here, and what value does it bring if there exists already example with `${kubernetes.pod.ip}`.

As you already have enough reviews - you can merge this PR